daffodil-japi/src/test/java/org/apache/daffodil/example/TestJavaAPI.java

```diff
@@ -1213,4 +1215,81 @@ public void testJavaAPI26() throws IOException, ClassNotFoundException, External
         assertTrue(DaffodilXMLEntityResolver.getXMLEntityResolver() != null);
         assertTrue(DaffodilXMLEntityResolver.getLSResourceResolver() != null);
     }
+
+    @Test
+    public void testJavaAPINullXMLTextEscapeStyle() throws IOException, ClassNotFoundException {
+        ByteArrayOutputStream xmlBos = new ByteArrayOutputStream();
+        try {
+            XMLTextInfosetOutputter outputter = new XMLTextInfosetOutputter(xmlBos, true, null);
+        } catch (Exception e) {
+            String msg = e.getMessage().toLowerCase();
+            assertTrue(msg.contains("unrecognized"));
+            assertTrue(msg.contains("null"));
+            assertTrue(msg.contains("xmltextescapestyle"));
+        }
+    }
+
+    @Test
+    public void testJavaAPICDATA1() throws IOException, ClassNotFoundException {
+        String expected = "<![CDATA[NO_WHITESPACE_AT_ALL]]>";
+        String data = "NO_WHITESPACE_AT_ALL$";
+        String schemaType = "string";
+        doXMLTextEscapeStyleTest(expected, data, schemaType);
```

Comment:
Currently the CDATA escape style always wraps all XML text content in CDATA brackets regardless of whether the text actually contains any whitespace or special characters.  I wonder if users would prefer the CDATA escape style to escape only XML text content which really needs to be wrapped in CDATA to preserve its original contents?  If such "smart" escaping would be more difficult to implement than it's worth, we can wait for users to request it and create an issue at that time.

Reply:
I think this is the right thing to do. It's worth the pass over the data before adding escaping of any kind. And this should only ever be needed for string data.